ble_app.h

```diff
@@ -0,0 +1,427 @@
+            return;
+        }
+
+        /* handle gap events */
+        _gap_handler.addEventHandler(this);
+        _ble.gap().setEventHandler(&_gap_handler);
+
+        /* This will inform us off all events so we can schedule their handling
+         * using our event queue */
+        _ble.onEventsToProcess(
+            makeFunctionPointer(this, &BLEApp::schedule_ble_events)
+        );
+
+        ble_error_t error = _ble.init(
+            this, &BLEApp::on_init_complete
+        );
+
+        if (error) {
+            print_error(error, "Error returned by BLE::init.\r\n");
+            return;
+        }
+
+        /* Process the event queue. */
+        _event_queue.dispatch_forever();
+        /* run left over events */
+        _event_queue.dispatch_once();
+
+        return;
+    }
+
+    /**
+     * Close advertising and/or existing connections and stop the App.
+     */
+    void stop()
+    {
+        /* let any left over events run */
+        _event_queue.call([this]() {
+            if (_ble.hasInitialized()) {
+                _ble.shutdown();
+                printf("Ble App stopped.\r\n");
+            }
+            _event_queue.break_dispatch();
+
+            _connected = false;
+            _is_connecting = false;
+            _is_scanning = false;
+            _gap_handler = ChainableGapEventHandler();
+        });
+    }
+
+    /**
+     * Subscribe with your own gap handler.
+     *
+     * @param[in] gap_handler Handler implementing selected ble::Gap::EventHandler methods.
+     *
+     * @returns True on success.
+     */
+    bool add_gap_event_handler(ble::Gap::EventHandler *gap_handler)
+    {
+        return _gap_handler.addEventHandler(gap_handler);
+    }
+
+    /** Set name we advertise as. */
+    bool set_advertising_name(const char *advertising_name)
+    {
+        char* new_name = nullptr;
+        size_t length = strlen(advertising_name) + 1;
+
+        if (advertising_name && length) {
+            new_name = (char*)malloc(length);
+            if (!new_name) {
+                return false;
+            }
+            memcpy(new_name, advertising_name, length);
+        }
+
+        _event_queue.call([this,new_name]() {
+            delete _advertising_name;
+            _advertising_name = new_name;
+            _event_queue.call([this]() { start_activity(); });
+        });
+
+        return true;
+    }
+
+    /** Set name we want to connect to. */
+    bool set_target_name(const char *target_name)
+    {
+        char* new_name = nullptr;
+        size_t length = strlen(target_name) + 1;
+
+        if (target_name && length) {
+            new_name = (char*)malloc(length);
+            if (!new_name) {
+                return false;
+            }
+            memcpy(new_name, target_name, length);
+        }
+
+        _event_queue.call([this,new_name]() {
+            delete _target_name;
+            _target_name = new_name;
+            _event_queue.call([this]() { start_activity(); });
```

Comment:
Same as above.

Reply:
That's how you start connecting - set the target name.

This is in case the app was already running when we set the name